ifrs17/Import/Importers.ipynb

```diff
@@ -485,19 +507,42 @@
         {
             "cell_type": "code",
             "source": [
-                "public async Task<ImportArgs[]> GetAllArgsAsync(ImportArgs primaryArgs)",
+                "public async Task<ImportArgs[]> GetAllArgsAsync(ImportArgs args, IDataSource dataSource, string format)",
                 "\n{",
-                "\n    if (primaryArgs.Scenario != null)",
-                "\n        return primaryArgs.RepeatOnce().ToArray();",
-                "\n",
-                "\n    var secondaryArgs = await DataSource.Query<PartitionByReportingNodeAndPeriod>() ",
-                "\n                               .Where(x => x.ReportingNode == primaryArgs.ReportingNode && x.Year == primaryArgs.Year && x.Month == primaryArgs.Month && x.Scenario != null)",
-                "\n                               .Select(x => new ImportArgs(x.ReportingNode, x.Year, x.Month, default(Periodicity), x.Scenario, ImportFormats.Cashflow)).ToArrayAsync();",
+                "\n    ImportArgs[] allArgs;",
+                "\n    switch(format)",
+                "\n    {",
+                "\n        case ImportFormats.YieldCurve : ",
+                "\n        {",
+                "\n            if(args.Scenario == null) {",
+                "\n                var scenariosWithYieldCurves = await dataSource.Query<YieldCurve>().Where(x => x.Year == args.Year && x.Month == args.Month && x.Scenario != null)",
+                "\n                    .Select(x => x.Scenario).Distinct().ToArrayAsync();",
+                "\n                var targetPartitions = await dataSource.Query<PartitionByReportingNodeAndPeriod>()",
+                "\n                    .Where(x => x.Year == args.Year && x.Month == args.Month && !scenariosWithYieldCurves.Contains(x.Scenario)).OrderBy(x => x.Scenario).ToArrayAsync();",
+                "\n                var targetScenarios = targetPartitions.Where(x => x.Scenario != null).Select(x => x.Scenario);",
+                "\n                if(targetScenarios.Any()) ApplicationMessage.Log(Warning.ScenarioReCalculations, String.Join(\", \", targetScenarios));",
+                "\n                allArgs = targetPartitions.Select(x => new ImportArgs(x.ReportingNode, x.Year, x.Month, default(Periodicity), x.Scenario, ImportFormats.Cashflow)).ToArray();",
+                "\n            }",
+                "\n            else {",
+                "\n                allArgs = (await dataSource.Query<PartitionByReportingNodeAndPeriod>()",
+                "\n                    .Where(x => x.Year == args.Year && x.Month == args.Month && x.Scenario == null).ToArrayAsync())",
```

Comment:
Why do we filter for Scenario null?
I understand this is the case of an import of YC for a scenario. Then I don't expect any recalculation for BE.

Reply:
Yes this is the case of an import of scenario YC. In fact there is no recalculation for BE. However, we need a list of all BE partitions for the target period in order to start calculations for all the resulting reporting nodes.